pointerevents/pointerevent_attributes_mouse-manual.html

```diff
@@ -17,68 +17,87 @@
+                        assert_true(event.buttons == 1, "If left mouse button is pressed buttons attribute is 1")
+                    }, pointerTestName + "'s buttons attribute is 1 when left mouse button is pressed.");
+                } else if (event.type == 'pointerup') {
+                    test(function() {
+                        assert_true(event.button == 0, "If left mouse button is just released button attribute is 0")
+                    }, pointerTestName + "'s button attribute is 0 when left mouse button is just released.");
+                    test(function() {
+                        assert_true(event.buttons == 0, "If left mouse button is just released buttons attribute is 0")
+                    }, pointerTestName + "'s buttons attribute is 0 when left mouse button is just released.");
+                } else {
+                    test(function() {
+                        assert_true(event.button == -1, "If mouse buttons are released button attribute is -1")
+                    }, pointerTestName + "'s button is -1 when mouse buttons are released.");
+                    test(function() {
+                        assert_true(event.buttons == 0, "If mouse buttons are released buttons attribute is 0")
+                    }, pointerTestName + "'s buttons is 0 when mouse buttons are released.");
+                }
+
+                // Test clientX and clientY
+                if (event.type != 'pointerout' && event.type != 'pointerleave' ) {
+                    test(function () {
+                        assert_true(event.clientX >= targetBoundingClientRect.left && event.clientX < targetBoundingClientRect.right, "ClientX should be in the boundaries of the black box");
+                    }, pointerTestName + ".clientX attribute is correct.");
+                    test(function () {
+                      assert_true(event.clientY >= targetBoundingClientRect.top && event.clientY < targetBoundingClientRect.bottom, "ClientY should be in the boundaries of the black box");
+                    }, pointerTestName + ".clientY attribute is correct.");
+                } else {
+                    test(function () {
+                        assert_true(event.clientX < targetBoundingClientRect.left || event.clientX > targetBoundingClientRect.right - 1 || event.clientY < targetBoundingClientRect.top || event.clientY > targetBoundingClientRect.bottom - 1, "ClientX/Y should be out of the boundaries of the black box");
+                    }, pointerTestName + "'s ClientX and ClientY attributes are correct.");
+                }
+
+                // Test isPrimary
+                test(function () {
+                    assert_equals(event.isPrimary, true, "isPrimary should be true");
+                }, pointerTestName + ".isPrimary attribute is correct.");
+
+                check_PointerEvent(event, testNamePrefix, pointerEventInterface);
+                detected_eventTypes[event.type] = true;
```

Comment:
Please move back <code>detected_eventTypes[event.type] = true<code> right after the <code>if</code> to the beginning of this method: makes the reentry-prevention easy to follow.